react/features/base/avatar/components/AbstractAvatar.js

```diff
@@ -0,0 +1,215 @@
+// @flow
+
+import { getGravatarURL } from 'js-utils/avatar';
+import { PureComponent } from 'react';
+
+import { getAvatarColor, getInitials } from '../functions';
+
+import { getParticipantById } from '../../participants';
+
+export type Props = {
+
+    /**
+     * URI of the avatar, if identifiable.
+     */
+    _avatarURL: ?string,
+
+    /**
+     * URI of the gravatar if identifyable.
+     */
+    _gravatarURL: ?string,
+
+    /**
+     * The string we base the initials on (this is generated from a list of precendences).
+     */
+    _initialsBase: ?string,
+
+    /**
+     * A string to override the initials to generate a color of. This is handy if you don't want to make
+     * the background color match the string that the initials are generated from.
+     */
+    colorBase?: string,
```

Comment:
Is this for the replaceement of AvtarListItem, so rooms in the same server get the same color? I find this and the initialsBase props quite confusing (but it's late so it may be that I'm tired).

Reply:
Yes, that is for the avatar list item, where we want to color things differently. The xxxBase postfix I use to denote something that some other thing will be "based on". E.g. initialsBase is something we base the initials on (e.g. this can be the name or the email). colorBase is the basis of the color, if provided